mlrun/api/db/sqldb/db.py

```diff
@@ -651,51 +639,92 @@ def list_tags(self, session, project: str):
         return tags
 
-    def add_project(self, session, project: dict):
-        project = project.copy()
-        name = project.get("name")
-        if not name:
-            raise ValueError("project missing name")
-
-        project.pop("users", [])
-        prj = Project(**project)
-        users = (
-            []
-        )  # self._find_or_create_users(session, user_names) - user_names are previously popped users
-        prj.users.extend(users)
-        self._upsert(session, prj)
-        self._projects.add(prj.name)
-        return prj.id
-
-    def update_project(self, session, name, data: dict):
-        prj = self.get_project(session, name)
-        if not prj:
-            raise DBError(f"unknown project - {name}")
-
-        data = data.copy()
-        data.pop("users", [])
-        for key, value in data.items():
-            if not hasattr(prj, key):
-                raise DBError(f"unknown project attribute - {key}")
-            setattr(prj, key, value)
-
-        users = (
-            []
-        )  # self._find_or_create_users(session, user_names) - user_names are previously popped users
-        prj.users.clear()
-        prj.users.extend(users)
-        self._upsert(session, prj, ignore=True)
-
-    def get_project(self, session, name=None, project_id=None):
-        if not (project_id or name):
-            raise ValueError("need at least one of project_id or name")
-
-        if project_id:
-            return self._query(session, Project).get(project_id)
-
-        return self._query(session, Project, name=name).one_or_none()
-
-    def delete_project(self, session, name: str):
+    def create_project(self, session: Session, project: schemas.Project):
+        self._create_project(session, project)
+
+    def _create_project(self, session: Session, project: schemas.Project):
+        logger.debug("Creating project in DB", project=project)
+        created = datetime.utcnow()
+        project.created = created
+        project = Project(
+            name=project.name,
+            description=project.description,
+            owner=project.owner,
+            source=project.source,
+            state=project.state,
+            created=created,
+            full_object=project.dict(),
+        )
+        self._upsert(session, project)
+
+    def store_project(self, session: Session, name: str, project: schemas.Project):
+        logger.debug("Storing project in DB", name=name, project=project)
+        project_record = self._get_project(session, name, raise_on_not_found=False)
+        if not project_record:
+            self._create_project(session, project)
+        else:
+            project.created = project_record.created
+            project_dict = project.dict()
+            project_record.full_object = project_dict
+            project_record.description = project.description
+            project_record.source = project.source
+            project_record.state = project.state
```

Comment:
* I prefer not changing the Project model in this PR, will add the column in a following PR
* same - not in this PR, and generally, created column already exists in field, I don't think it worth a migration just for the rename, I might rename the schema itself (what's returned from the API). I'm not sure I don't prefer something like `creation_time`
* It's not `_generate_project_record...` since it's overriding the existing one, so maybe `_update_project_record...`
TBH this is a 12 lines method so I don't see why that's needed but ok